providers/base/bin/wwan_tests.py

```diff
@@ -362,6 +362,45 @@ def invoked(self):
         sys.exit(ret_code)
 
 
+class ThreeGppScanTest:
+
+    def register_argument(self):
+
+        parser = argparse.ArgumentParser()
+        parser.add_argument(
+            "hw_id", type=str, help="The hardware ID of the modem"
+        )
+        parser.add_argument(
+            "--timeout",
+            type=int,
+            default=300,
+            help="timeout for 3gpp-scan",
+        )
+        return parser.parse_args(sys.argv[2:])
+
+    def invoked(self):
+
+        args = self.register_argument()
+        ret_code = 1
+        try:
+            mm = MMCLI()
+            mm_id = mm.equipment_id_to_mm_id(args.hw_id)
+            _wwan_radio_on()
+            cmd = [
+                "mmcli",
+                "-m",
+                str(mm_id),
+                "--3gpp-scan",
+                "--timeout",
+                str(args.timeout),
+            ]
+            ret_code = subprocess.check_call(cmd)
```

Comment:
Using `check_call`, you will only get back the return code of the `mmcli` command. It's fine if there is no useful feedback we could get from the output, but otherwise it might be interesting to capture the output in case the command fails. (I forgot what output `mmcli --3gpp-scan` provides, so I'm not sure if my comment makes sense :))

Reply:
Here is some errors we might seen during the scanning network test, so we probably could check is `error` shown in the log.
```
# the modem is not available
admin@C0310A2043-00843:~$ sudo mmcli -m 0 --3gpp-scan --timeout 200; echo $?
error: couldn't find modem
1

# the modem is not enabled
admin@C0310A2043-00843:~$ sudo mmcli -m 1 --3gpp-scan --timeout 200; echo $?
error: modem not enabled yet
1

# Unable to scan available network within a period of time
admin@C0310A2043-00843:~$ sudo mmcli -m 1 --3gpp-scan --timeout 30; echo $?
error: couldn't scan networks in the modem: 'Timeout was reached'
1
```